pkg/server/container_create.go

```diff
@@ -137,6 +119,39 @@ func (c *criContainerdService) CreateContainer(ctx context.Context, r *runtime.C
 		}
 	}()
 
+	// Create container volumes mounts.
+	// TODO(random-liu): Add cri-containerd integration test for image volume.
+	volumeMounts := generateVolumeMounts(containerRootDir, image.Config)
+
+	// Generate container runtime spec.
+	mounts := c.generateContainerMounts(getSandboxRootDir(c.rootDir, sandboxID), config)
+
+	spec, err := c.generateContainerSpec(id, sandboxPid, config, sandboxConfig, image.Config, append(mounts, volumeMounts...))
```

Comment:
I think you need to subtract from `volumeMounts` anything which is also expliclty in `mounts`, since an explicit bind mount is supposed to take precedence over the implicit ones.

Reply:
BTW -- this would also mean that as soon as Kubelet starts to understand these things and provide them via `mounts` directly it will magically stop trying to do things here.

Reply:
> I think you need to subtract from volumeMounts anything which is also expliclty in mounts, since an explicit bind mount is supposed to take precedence over the implicit ones.

Done.

> BTW -- this would also mean that as soon as Kubelet starts to understand these things and provide them via mounts directly it will magically stop trying to do things here.

We'll see how CRI will change in that case, and make change our side correspondingly. Added TODO for it.